wagtail/admin/checks.py

```diff
@@ -164,3 +164,22 @@ def check_panels_in_model(cls, context="model"):
         errors.append(error)
 
     return errors
+
+
+@register("wagtail_admin_base_url")
+def wagtail_admin_base_url_check(app_configs, **kwargs):
+    from django.conf import settings
+
+    errors = []
+
+    if getattr(settings, "WAGTAILADMIN_BASE_URL", None) is None:
+        errors.append(
+            Warning(
+                "The WAGTAILADMIN_BASE_URL setting is not defined",
+                hint="This should be the base URL used to access the Wagtail admin site. "
+                "Without this, URLs in notification emails will not display correctly",
+                id="wagtailadmin.W003",
+            )
+        )
+
```

Comment:
will we need to add something to upgrade considerations?

Reply:
I think https://docs.wagtail.org/en/stable/releases/3.0.html#base-url-setting-renamed-to-wagtailadmin-base-url probably covers everything we need to say already - this warning just makes it more visible.